### DISCUSSION TITLE
chakra-ui目前支持按需加载吗？

### DISCUSSION THREAD

**Discussion by 731514100** [chakra-ui/chakra-ui#8326] [Ideas] [answered]
我找了找资料，并没有发现有按需加载的实现方案。我的项目中除了Box、Text、Grid、GridItem、Flex、Center这些组件在使用，其他组件并没有使用到，就非常想剔除那些没有加载的组件，但是目前没有什么办法。如果我项目是B端项目，我也不会纠结这个事情，因为目前项目是C端项目，对性能指标有一定需求，想问问各位大佬们有实现的方案吗？

**Comment by segunadebayo** [comment, marked as answer]
v3 ships as one package. Unused components get dropped by the bundler if you import them as named exports.

You do not need a babel plugin or a separate `chakra-ui/xxx` path. Import only `Box`, `Text`, and `Flex` from `@chakra-ui/react` and tree-shaking does the rest.
